enumerables.rb

```diff
@@ -0,0 +1,137 @@
+module Enumerable
+  def my_each
+    i = 0
+    arr = self
+    if block_given?
+      while i < arr.count
+        yield(arr[i])
+        i += 1
+      end
+    else
+      arr
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    arr = self
+    if block_given?
+      while i < arr.count
+        yield(arr[i], i)
+        i += 1
+      end
+    else
+      arr
```

Comment:
- [ ] Same issue as my_each here. Return an Enumerator, not just the self. You could try to find a way to convert the self to an Enumerator for example.